### PR TITLE
chore: cow amm module deployments

### PR DESCRIPTION
**Pull request by mfw78**
This PR:

1. Publishes script outputs from deployment of the `CowAmmModule`
2. Updates the `network.json` with deployment addresses.
3. Includes the audit from `G0` for the `CowAmmModule`

To test:

Verify that that deployment addresses contain verified smart contract code related to the previous commit on f075b10.

### CODE REVIEW

**Review by fleupold — APPROVED** [review]
No counterfactual deployment to have the same address on all chains?

**Comment by fedgiac** [comment]
Waiting for finalizing the comments in #33 before reviewing.

> No counterfactual deployment to have the same address on all chains?

I favored non-deterministic addresses in this project because (more or less) every time you add o remove a new dependency in Foundry it changes the deterministic deployment address, even if the dependency is unused. (This is because the repo's remappings are hashed in in the metadata.)
Overall, dependencies in Foundry are a mess and I wanted to sort them out before binding ourselves to deterministic addresses (which would force us to a more painful deployment process in the future for other chains, since we'd have to pick the right commit and synchronize/add/delete git submodules).

**Comment by mfw78** [comment]
@fedgiac Ready for review :pray:

File: broadcast/CowAmmModule.s.sol/1/run-1709653244.json [hunk 1]
```diff
@@ -0,0 +1,53 @@
+{
+  "transactions": [
+    {
+      "hash": "0x3720c9ae34c99233312375bc50528f24c8134c65040490f504617a7f6aa803b6",
```
Comment (fedgiac) [inline broadcast/CowAmmModule.s.sol/1/run-1709653244.json:4]: I'm unable to locate this transaction hash. I believe this is the drawback of using a forked node for resimulating. :cry:.

Comment (fedgiac) [reply broadcast/CowAmmModule.s.sol/1/run-1709653244.json:4]: [This](https://etherscan.io/tx/0x99eafdf186e890a29196aa14223bbf7a0a59ecbcdbd9f5bf6314de90e78e4161) appears to be the right transaction hash. Do you still have the run file with this hash?

**Review by fedgiac — APPROVED** [review]

**Pull request merged**